tests/ui/vec_box_sized.rs

```diff
@@ -33,6 +59,10 @@ mod should_not_trigger {
         // Regression test for #3720. This was causing an ICE.
         inner: Vec<Box<T>>,
     }
+
+    fn allocator_mismatch() -> Vec<Box<i32, DummyAllocator>> {
+        Vec::new()
+    }
```

Comment:
Could you add a second test, where the allocator of the vector mismatches, like his:
```
    fn allocator_mismatch() -> Vec<Box<i32>, DummyAllocator> {
        Vec::new()
    }
```